collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/core/client/testexecution/TestExecutionClientImpl.java

```diff
@@ -322,17 +323,25 @@ private TestCase createTestCase(TestExecution.Test test, TestRun testRun, Featur
                 testCase.setStatus(TestCaseStatus.Unknown);
             }
 
-             Set<String> tags = getStoryIds(feature.getIssueLinks());
-            // Temporarily commented for core project update
-            // testCase.setTags(tags);
             testCase.setTestSteps(this.getTestSteps(testRun));
+            List<Feature> tc = featureRepository.getStoryByNumber(test.getKey());
+            if (!CollectionUtils.isEmpty(tc)) {
+                for (Feature t : tc) {
+                    Set<String> tags = getStoryIds(t.getIssueLinks());
+                    testCase.setTags(tags);
+                }
+            }
         }
         return testCase;
     }
 
     private Set<String> getStoryIds(Collection<FeatureIssueLink> issueLinks) {
         Set<String> tags = new HashSet<>();
-        issueLinks.forEach(issueLink -> tags.add(issueLink.getTargetIssueKey()));
+        issueLinks.forEach(issueLink -> {
+            if(issueLink.getIssueLinkType().equalsIgnoreCase("tests")) {
```

Comment:
use StringUtils to make the comparison null safe

Reply:
reformatted to be null safe